src/gui/qgsmaplayerstylecategoriesmodel.cpp

```diff
@@ -65,6 +65,16 @@ void QgsMapLayerStyleCategoriesModel::setCategories( QgsMapLayer::StyleCategorie
   if ( mCategories == categories )
     return;
 
+  // determine the allowed categories to consider only these in the last categories.
```

Comment:
I'm having a hard time to understand this comment. I assume "last" is about the "last used category selection" which is a potential caller of this method, but within this method we cannot assume that we are applying last used.

```suggestion
  // filter the categories and only preserve the categories compatible with the current layer type
```

If I understand correctly?

Reply:
yes you understood correctly. thanks :-)